Jieun/services/userService.js

```diff
@@ -1,28 +1,39 @@
 const userDao = require("../models/userDao");
+const {
+  passwordValidationCheck,
+  emailValidationCheck,
+} = require("../utils/validation-check");
 const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
 
 const signUp = async (name, profileImage, email, password, phoneNumber) => {
-  const pwValidation = new RegExp(
-    "^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})"
-  );
-  if (!pwValidation.test(password)) {
-    const err = new Error("PASSWORD_IS_NOT_VALID");
-    err.statusCode = 409;
-    throw err;
-  }
+  await passwordValidationCheck(password);
+  await emailValidationCheck(email);
+
   const saltRounds = 10;
   const hashedPassword = await bcrypt.hash(password, saltRounds);
-  const createUser = await userDao.createUser(
+  return userDao.createUser(
     name,
     profileImage,
     email,
     hashedPassword,
     phoneNumber
   );
+};
 
-  return createUser;
+const login = async (email, password) => {
+  const [hashedPassword] = await userDao.loginUser(email);
+  const result = await bcrypt.compare(password, hashedPassword.password);
+  if (!result) {
+    const err = new Error("Invalid User");
+    err.statusCode = 409;
+    throw err;
+  }
+  const payLoad = { userId: hashedPassword.id };
+  return jwt.sign(payLoad, process.env.secretKey);
```

Comment:
```suggestion
  const [user] = await userDao.getUserByEmail(email);
  
  if (!user || !bcrypt.compare(password, user.password)) throw new Error("Invalid Email or Password");
 
  return jwt.sign({ userId: user.id }, process.env.JWT_SECRET);
```
1. password 뿐만 아니라 email이 일치하지 않는 경우에도 핸들링이 필요합니다.
2. payload 같은경우는 현재 많은 데이터가 들어가지 않고, userId만 들어가기에 변수선언 없이 곧바로 작성되어도 깔끔해보입니다. 
3. 환경변수에서 읽어오는 값은 상수 데이터이므로 대문자를 통해 작성되어야 합니다!